proposals/0010-resource-element-type-validation.md

```diff
@@ -0,0 +1,160 @@
+* Proposal: [0010](0010-resource-element-type-validation.md)
+* Author(s): [Joshua Batista](https://github.com/bob80905)
+* Sponsor: Joshua Batista
+* Status: **Under Consideration**
+* Impacted Project(s): (LLVM)
+
+* Issues: [#75676](https://github.com/llvm/llvm-project/issues/75676)
+
+## Introduction
+Resources are often used in HLSL, with various resource element types (RETs).
+For example:
+```
+RWBuffer<float> rwbuf: register(u0);
+```
+In this code, the RET is `float`, and the resource type is `RWBuffer`.
+There are two types of buffers, RawBuffers and TypedBuffers. `RWBuffer`
+is a TypedBuffer variant, and `StructuredBuffer` is a RawBuffer variant.

```

Comment:
Given a random buffer type, how do I determine if it is raw or typed?  I really think just having a list of the HLSL data types and whether they're raw or typed would help here.  I don't even think it is long, but obfuscating it in prose doesn't help.

* `[Append|Consume|RW]StructuredBuffer` - raw
* `[RW]Buffer` - typed
* `[RW]ByteAddressBuffer` - raw?
* `[Feedback]Texture*` - typed (I assume Feedback textures are typed)
*